torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -238,6 +247,18 @@ void OwnerRRef::setValue(IValue&& value) {
   }
 }
 
+void OwnerRRef::setError(const std::string& error) {
+  std::unique_lock<std::mutex> lock(mutex_);
+  error_ = error;
+  std::shared_ptr<FutureMessage> future;
+  future.swap(future_);
+  lock.unlock();
+  valueCV_.notify_all();
+  if (future.get()) {
+    future->setErrorIfNeeded(error);
```

Comment:
This means we are not going to throw away the error when the future is already complete. It might need more discussion to figure out what would be the appropriate solution to handle this. For now, maybe let's log in `setErrorIfNeeded` to indicate that an error occurred after completion?